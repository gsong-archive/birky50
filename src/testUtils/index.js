import * as faker from "faker";

const address = () =>
  [
    faker.address.streetAddress(),
    faker.address.city(),
    `${faker.address.stateAbbr()} ${faker.address.zipCode()}`,
  ].join(", ");

const card = () => ({
  name: `${faker.name.firstName()} ${faker.name.lastName()}`,
  url: faker.internet.url(),
  price: faker.random.word(),
  address: address(),
  addressUrl: faker.internet.url(),
  phone: faker.phone.phoneNumber(),
  imgUrl: faker.image.imageUrl(),
});

const generate = {
  address,
  card,
  phoneNumber: faker.phone.phoneNumber,
  url: faker.internet.url,
  words: faker.random.words,
};

export { generate };
