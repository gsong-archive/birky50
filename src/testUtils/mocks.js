import * as cssFeatures from "../styles/cssFeatures";
import * as styleUtils from "../styles/utils";

export const mockGetBaseFontSize = {
  mock: () => (styleUtils.getBaseFontSize = jest.fn(() => 16)),
  restore: () => styleUtils.getBaseFontSize.mockRestore(),
};

export const mockSupportsGrid = bool => (cssFeatures.supportsGrid = bool);

export const mockSupportsObjectFit = bool =>
  (cssFeatures.supportsObjectFit = bool);
