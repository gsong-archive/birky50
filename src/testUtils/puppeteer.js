import "../env";

const DEFAULT_APP_URL = "http://localhost:3000";
const DEFAULT_JEST_TIMEOUT = 5000;
const DEFAULT_SCREENSHOTS_FOLDER = "screenshots/puppeteer";
const DEFAULT_SCROLL_CHECK_DELAY = 100;

const appUrl =
  process.env.APP_URL === undefined ? DEFAULT_APP_URL : process.env.APP_URL;

const jestTimeOut =
  parseInt(process.env.JEST_TIMEOUT, 10) || DEFAULT_JEST_TIMEOUT;

const screenshotsFolder =
  process.env.PUPPETEER_SCREENSHOTS_FOLDER === undefined
    ? DEFAULT_SCREENSHOTS_FOLDER
    : process.env.PUPPETEER_SCREENSHOTS_FOLDER;

const scrollCheckDelay =
  process.env.PUPPETEER_SCROLL_CHECK_DELAY || DEFAULT_SCROLL_CHECK_DELAY;

export const settings = { appUrl, jestTimeOut, scrollCheckDelay };

export const screenshot = async (imageName, page, options = {}) => {
  await page.screenshot({
    path: `${screenshotsFolder}/${imageName}.png`,
    ...options,
  });
};

const desktops = {
  "macbook-13": { width: 1280, height: 800 },
  "macbook-15": { width: 1440, height: 900 },
};

const mobiles = {
  "ipad-portrait": { width: 768, height: 1024 },
  "iphone-se-portrait": { width: 320, height: 568 },
  "iphone-8-portrait": { width: 375, height: 667 },
  "iphone-8-plus-portrait": { width: 540, height: 960 },
  "iphone-x-portrait": { width: 375, height: 812 },
};

Object.entries(mobiles).forEach(([device, viewport]) => {
  mobiles[device.replace("portrait", "landscape")] = {
    width: viewport.height,
    height: viewport.width,
    isLandscape: true,
  };
});

Object.entries(mobiles).forEach(([device, viewport]) => {
  mobiles[device] = { ...viewport, isMobile: true, hasTouch: true };
});

export const devices = Object.assign({}, desktops, mobiles);
