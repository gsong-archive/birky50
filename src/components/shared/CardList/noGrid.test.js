//@flow
import React from "react";

import { render } from "react-testing-library";

import { generate } from "testUtils";
import { mockGetBaseFontSize, mockSupportsGrid } from "testUtils/mocks";

mockGetBaseFontSize.mock();
mockSupportsGrid(false);

const CardList = require("./index").default;

describe("CardList renders", () => {
  const props = {
    items: [...Array(10)].map(() => generate.card()),
  };
  const { container } = render(<CardList {...props} />);

  test("cards are visible", () =>
    props.items.forEach(item =>
      expect(container).toHaveTextContent(item.name)
    ));

  test("CardList is rendered", () => {
    const props = {
      items: [
        {
          name: "ABC Co.",
          url: "https://som_url/",
          price: "$100â€“$200",
          address: "123 Main Street",
          addressUrl: "https://url_to_map/",
          phone: "415-123-4567",
          imgUrl: "https://url_to_image/",
        },
      ],
    };
    const { container } = render(<CardList {...props} />);
    expect(container).toMatchSnapshot();
  });
});

mockGetBaseFontSize.restore();
