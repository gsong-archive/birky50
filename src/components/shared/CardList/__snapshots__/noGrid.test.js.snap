// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList renders CardList is rendered 1`] = `
.emotion-19 {
  display: grid;
  grid-gap: 1.5rem;
  margin-bottom: 0.75em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-19 > div {
  margin-right: 1.5rem;
  margin-bottom: 1.5rem;
}

.emotion-19 > div {
  -webkit-flex: 0 0 21.25rem;
  -ms-flex: 0 0 21.25rem;
  flex: 0 0 21.25rem;
}

.emotion-16 {
  margin: 0.75rem;
}

.emotion-6 {
  margin: 0.75rem 0 0;
  color: hsl(0,0%,45%);
}

.emotion-15 > * {
  margin-top: 0.75rem;
}

.emotion-17 {
  border-radius: 0.25rem;
  box-shadow: 0 2px 6px 0 hsla(0,0%,0%,0.2);
}

.emotion-4 {
  font-weight: bold;
}

.emotion-2 {
  color: hsl(44,56%,32%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.emotion-7 {
  font-family: "Segoe UI Emoji";
  margin-right: 0.25em;
}

.emotion-0 {
  border-radius: 0.25rem;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
  display: block;
  height: 12rem;
  background-image: url(https://url_to_image/);
  background-size: cover;
  background-position: center center;
}

<div>
  <div
    class="emotion-19 emotion-20"
  >
    <div
      class="emotion-17 emotion-18"
    >
      <a
        href="https://som_url/"
        rel="nofollow"
        class="emotion-0 emotion-1"
      />
      <div
        class="emotion-16"
      >
        <div
          class="emotion-4 emotion-5"
        >
          <a
            href="https://som_url/"
            rel="nofollow"
            class="emotion-2 emotion-3"
          >
            ABC Co.
          </a>
        </div>
        <p
          class="emotion-6"
        >
          $100‚Äì$200
        </p>
        <address
          class="emotion-15"
        >
          <div>
            <span
              class="emotion-10"
            >
              <span
                class="emotion-7"
                role="img"
                aria-label="round pushpin"
              >
                üìç
              </span>
              <a
                href="https://url_to_map/"
                class="emotion-2 emotion-3"
              >
                123 Main Street
              </a>
            </span>
          </div>
          <div>
            <span
              class="emotion-10"
            >
              <span
                class="emotion-7"
                role="img"
                aria-label="phone receiver"
              >
                üìû
              </span>
              <a
                href="tel:415-123-4567"
                class="emotion-2 emotion-3"
              >
                415-123-4567
              </a>
            </span>
          </div>
        </address>
      </div>
    </div>
  </div>
</div>
`;
